src/clr-angular/data/datagrid/providers/selection.ts

```diff
@@ -124,6 +129,7 @@ export class Selection {
 
   public clearSelection(): void {
     this.current.length = 0;
+    this.prevSelectionRefs = [];
```

Comment:
Approving but would like a test for this.

Reply:
I don't have a super direct way of testing this bug fix, but added a test case that I think is ok-ish. Take a look and let me know if you think there is a better way.